roles/fairfood/templates/post-receive.j2

```diff
@@ -53,6 +56,13 @@ upgrade_application() {
   echo "Restart puma..."
   bundle exec pumactl restart # Hot restart helps reduce downtime
 
+  echo "Restarting job workers..."
+  systemctl --user restart "jobs-{{ app }}.service"
```

Comment:
Restart performs a stop and start. The stop will send the TERM signal to SolidQueue again if it's still running (we sent it at the beginning of the deploy already). It's unlikely though because the default configuration waits only 5 seconds for jobs to finish. The deployment usually takes longer than 5 seconds. After that, jobs are marked as failed, put back in the queue and picked up again after the restart.

We may want a longer timeout for jobs. I think that we should measure the time it takes to deploy and wait at least that long because it doesn't delay us anyway. We may want to call this later as well. The generation of the error pages takes a bit of time because they need to wait for puma to properly boot up and be ready.

We may want to

Reply:
Nice, I like the idea of letting jobs finish gracefully. We almost want it to wait asynchronously for all jobs to finish, and only then restart. If a large report was running over several minutes, then SolidQueue would not start any new jobs, and restart when ready. 
But if SQ doesn't have that ability built in, then I think this is a great solution. 
* I agree that we can give the graceful stop a longer timer.
* And also agree to move the restart until after the page generations, which gives any longrunning tasks a bit more grace period.